store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedAddGraphHandlerTest.java

```diff
@@ -429,4 +435,32 @@ public void shouldAddGraphWithACache() throws StoreException, OperationException
         }
         assertThat(set).contains(EXPECTED_GRAPH_ID, EXPECTED_GRAPH_ID_2);
     }
+
+    @Test
+    public void shouldAddGraphWithHandler() throws Exception {
+        store.initialise(FEDERATEDSTORE_GRAPH_ID, null, federatedStoreProperties);
+        final Schema expectedSchema = new Schema.Builder().build();
+
+        final AccumuloProperties clone = PROPERTIES.clone();
+
+        clone.set(OPERATION_DECLARATIONS_JSON, new String(JSONSerialiser.serialise(new OperationDeclarations.Builder()
+                .declaration(new OperationDeclaration.Builder()
+                        .handler(new TestErrorHandler())
+                        .operation(GetAllElements.class)
+                        .build())
+                .build(), true)));
+
+        new FederatedAddGraphHandler().doOperation(
+                new AddGraphWithHooks.Builder()
+                        .graphId(EXPECTED_GRAPH_ID)
+                        .schema(expectedSchema)
+                        .storeProperties(clone)
+                        .hooks()
+                        .build(),
+                new Context(testUser),
+                store);
+
+        assertThatThrownBy(() -> store.execute(new GetAllElements(), new Context(testUser)))
```

Comment:
Usually nicer to use `assertThatExceptionOfType(ExpectedType).isThrownBy(....)`

Reply:
Sure I don't mind either way. I'm just pleased the older way of doing this is gone.